lib/secure_headers/headers/content_security_policy.rb

```diff
@@ -212,5 +212,13 @@ def strip_source_schemes(source_list)
     def symbol_to_hyphen_case(sym)
       sym.to_s.tr("_", "-")
     end
+
+    def source_scheme(source)
+      uri = URI(source.sub(PORT_WILDCARD_REGEX, ""))
+      # If host is nil the given source doesn't contain a scheme
+      # e.g. for `example.org:443` it would return `example.org` as the scheme
+      # which is of course incorrect
+      uri.scheme if uri.host
```

Comment:
It turns out that this also fails for URLs like `https://*.example.org:*`

What do you think of hardcoding just HTTP and HTTPS to avoid further wildcard issues?
```suggestion
      return "https" if source.start_with?("https://")
      return "http" if source.start_with?("http://")
      return nil
```

Reply:
Yeah I agree. I'll add a test case for that today and change it to the regex you suggested.

Reply:
Actually having looked into it again I don't quite get what the issue is. Parsing `https://*.example.org:*` would indeed fail but this is why in line 217 `uri = URI(source.sub(PORT_WILDCARD_REGEX, ""))` it removes the port wildcard.

```
irb(main):003:0> PORT_WILDCARD_REGEX = /:\*/
=> /:\*/
irb(main):004:0> source = "https://*.example.org:*"
=> "https://*.example.org:*"
irb(main):005:0> URI(source.sub(PORT_WILDCARD_REGEX, ""))
=> #<URI::HTTPS https://*.example.org>
irb(main):006:0> _.scheme
=> "https"
```

Reply:
FYI, I've tried to give this whole code a refactor at https://github.com/github/secure_headers/pull/497